picard/ui/profileeditor.py

```diff
@@ -0,0 +1,455 @@
+
+    def get_settings_for_profile(self, id):
+        """Get the settings for the specified profile ID.
+
+        Args:
+            id (str): ID of the profile
+
+        Returns:
+            dict: Profile settings
+        """
+        # Add empty settings dictionary if no dictionary found for the profile.
+        if id not in self.profile_settings:
+            self.profile_settings[id] = {}
+        return self.profile_settings[id]
+
+    def get_current_selected_item(self):
+        """Gets the profile item currently selected in the profiles list.
+
+        Returns:
+            ProfileListWidgetItem: Currently selected item
+        """
+        items = self.ui.profile_list.selectedItems()
+        if items:
+            return items[0]
+        return None
+
+    def update_current_expanded_items_list(self):
+        if self.building_tree:
+            return
+        self.expanded_sections = []
+        for i in range(self.ui.settings_tree.topLevelItemCount()):
+            tl_item = self.ui.settings_tree.topLevelItem(i)
+            if tl_item.isExpanded():
+                self.expanded_sections.append(tl_item.text(0))
+
+    def profile_selected(self, update_settings=True):
+        """Update working profile information for the selected item in the profiles list.
+
+        Args:
+            update_settings (bool, optional): Update settings tree. Defaults to True.
+        """
+        item = self.get_current_selected_item()
+        if item:
+            id = item.profile_id
+            self.current_profile_id = id
+            settings = self.get_settings_for_profile(id)
+            if update_settings:
+                self.make_setting_tree(settings=settings)
+        else:
+            self.current_profile_id = None
+            self.make_setting_tree(settings=None)
+
+    def make_setting_tree(self, settings=None):
+        """Update the profile settings tree based on the settings provided.
+        If no settings are provided, displays an empty tree.
+
+        Args:
+            settings (dict, optional): Dictionary of settings for the profile. Defaults to None.
+        """
+        self.set_button_states()
+        self.ui.settings_tree.clear()
+        self.ui.settings_tree.setHeaderItem(QtWidgets.QTreeWidgetItem())
+        self.ui.settings_tree.setHeaderLabels([_("Settings to include in profile")])
+        if settings is None:
+            return
+        self.building_tree = True
+        for id, group in UserProfileGroups.SETTINGS_GROUPS.items():
+            title = group["title"]
+            group_settings = group["settings"]
+            widget_item = QtWidgets.QTreeWidgetItem([title])
+            widget_item.setFlags(QtCore.Qt.ItemIsEnabled | QtCore.Qt.ItemIsUserCheckable | QtCore.Qt.ItemIsAutoTristate)
+            widget_item.setCheckState(0, QtCore.Qt.Unchecked)
+            for setting in group_settings:
+                child_item = QtWidgets.QTreeWidgetItem([_(setting[1])])
```

Comment:
I don't like numeric indexes.
We could use namedtuple for profile settings, as per https://github.com/rdswift/picard/pull/9